pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -84,10 +85,13 @@ func (cd *clusterDeploy) sync(key string, cluster *v3.Cluster) (runtime.Object,
 		} else {
 			cluster.Spec.RancherKubernetesEngineConfig.Authentication.Strategy = "x509"
 		}
+		logrus.Debugf("clusterDeploy: sync: cluster.Spec.RancherKubernetesEngineConfig.Authentication.Strategy set to [%s] for cluster [%s]", cluster.Spec.RancherKubernetesEngineConfig.Authentication.Strategy, cluster.Name)
 	}
 
 	err = cd.doSync(cluster)
 	if cluster != nil && !reflect.DeepEqual(cluster, original) {
+		logrus.Debugf("clusterDeploy: sync: original [%v]", original)
+		logrus.Debugf("clusterDeploy: sync: cluster [%v]", cluster)
```

Comment:
Since we are logging entire cluster objects, I feel we should have this info at TRACE level. This is too verbose to have at debug?

Reply:
We don't use TRACE in 2.x, but I can file an issue to make sure we can use that for more extensive logging. There are more full file content dumps in this PR (k8s yaml and kubeconfig).

Currently the backend of our logserver/loglevel (logrus) does support multiple levels, we just have to enable them in our tools and probably, while we are at it, make a scheme for what goes where so everyone knows what to do.

Also this needs to be communicated so people know it can go in TRACE (as this is not possible today afaik)